packages/core/src/helpers/plugable-record.ts

```diff
@@ -0,0 +1,26 @@
+const type = Symbol();
+
+export type PlugableRecord = Record<Key, unknown>;
+export type Key<T = unknown> = symbol & { [type]: T };
+export type Val<K extends Key> = K[typeof type];
+
+export const plugableRecord = {
+  key<T>() {
+    return Symbol() as Key<T>;
+  },
+  set<K extends Key>(map: PlugableRecord, key: K, value: Val<K>) {
+    map[key] = value;
+  },
+  get<K extends Key>(map: PlugableRecord, key: K): Val<K> | undefined {
+    return map[key];
+  },
+  getAssure<K extends Key>(map: PlugableRecord, key: K): Val<K> {
+    if (!map[key]) {
+      throw new Error(`key ${key.description} is missing on map`);
+    }
+    return map[key];
+  },
+  getUnsafeAssure<K extends Key>(map: PlugableRecord, key: K): Val<K> {
```

Comment:
```suggestion
  getUnsafeAssure<K extends Key>(map: PlugableRecord, key: K): Val<K> | undefined {
```
?

Reply:
thats the `unsafe` part of the method, `get` returns `Val<K> | undefined` and `getAssure` returns `Val<K>` or throws

Reply:
I understand what it does, but `getUnsafeAssure` return type will always be `Val<K>` right now.

Reply:
it is used by [places](https://github.com/wix/stylable/pull/2125/files#diff-1936f22036b3d48f2edc918dc1d252051ca2155259914cbe7f69412bda43cf09R54) that know for a fact that the value is set and allows them to ignore undefined check for type and runtime value

Reply:
I changed the name to `getUnsafe`, which might be a bit more clear, what do you think?